libs/island-ui/core/src/lib/Tabs/Tabs.tsx

```diff
@@ -64,9 +70,19 @@ export const Tabs: FC<TabInterface> = ({
     setIsMobile(false)
   }, [width])
 
-  if (isDefined(tab?.currentId) && prevCurrentId !== tab.currentId) {
-    onChangeHandler?.(tab?.currentId)
-    setPrevCurrentId(tab.currentId)
+  useEffect(() => {
```

Comment:
Changed it to useEffect since it was producing error both with onlyRenderSelectedTab and without.

WIth onlyRenderSelectedTab
<img width="914" alt="image" src="https://user-images.githubusercontent.com/112904566/202207722-4a40821d-9d51-4931-ad08-fc399cc7a2b3.png">

WIthout onlyRenderSelectedTab

![Uploading image.png…]()

Note the conditional inside the useEffect runs only on tab click